Vic2ToHoI4/Source/HOI4World/Names/CultureNames.cpp

```diff
@@ -0,0 +1,72 @@
+#include "CultureNames.h"
+#include "OSCompatibilityLayer.h"
+#include "ParserHelpers.h"
+
+
+
+HoI4::CultureNames::CultureNames(std::istream& theStream)
+{
+	registerKeyword("first_names", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		maleNames = nameStrings.getStrings();
+	});
+	registerKeyword("last_names", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		surnames = nameStrings.getStrings();
+	});
+	registerKeyword("female_names", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		femaleNames = nameStrings.getStrings();
+	});
+	registerKeyword("female_surnames", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		femaleSurnames = nameStrings.getStrings();
+	});
+	registerKeyword("callsigns", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		callsigns = nameStrings.getStrings();
+	});
+	registerKeyword("industry_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		industryCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("electronic_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		electronicCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("car_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		carCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("naval_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		navalCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("weapon_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		weaponCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("aircraft_companies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		aircraftCompanies = nameStrings.getStrings();
+	});
+	registerKeyword("intelligence_agencies", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList nameStrings(theStream);
+		intelligenceAgencies = nameStrings.getStrings();
+	});
+	registerRegex("[A-Za-z0-9\\_]+", commonItems::ignoreItem);
+
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+
+void HoI4::CultureNames::convertNamesToUTF8()
+{
+	std::for_each(maleNames.begin(), maleNames.end(), [](std::string& name) {
+		name = Utils::convertWin1252ToUTF8(name);
```

Comment:
Please confirm you're actually loading 1252 names. I seem to recall HoI4 was UTF8 in its files.

Reply:
Unless these are vic2 names. In which case, sigh, one day you'll have to have a __APPLE__ that would skip conversion.

Reply:
That's exactly it. I renamed the function to be more clear.